docs/src/chapters/testy.adoc

```diff
@@ -1,12 +1,262 @@
+[source, javascript]
+----
+include::../../../webapp/tests/components/common/Navbar.test.tsx[lines=6..12]
+----
+
+Kolejny test zaprezentowany poniżej odpowiedzialny jest za sprawdzenie formularza zmiany hasła. Po wprowadzeniu
+prawidłowych danych oczekiwany jest komunikat o powodzeniu operacji.
+
+.Test sprawdzający formularz zmiany hasła z pliku PasswordChangePage.test.java
+[source, javascript]
+----
+include::../../../webapp/tests/pages/userPanel/account/PasswordChangePage.test.tsx[lines=25..42]
+----
+
+Kod napisany w _Typescript_ został pokryty testami w ponad 65%. Stworzono 32 przypadki testowe, zawierające 93 testy.
+Jest to rezultat widocznie gorszy niż w przypadku części serwerowej. Pewien wpływ miały na to zapewne opóźnienia w pracy
+nad projektem i brak dodatkowego czasu na implementację kolejnych testów.
+
+.Raport z pokrycia testami kodu części frontendowej
+image::../images/tests_chapter/test_coverage_frontend.png[]
+
+==== Aplikacja mobilna
+
+Testy dla aplikacji mobilnej, podobnie jak w przypadku części frontendowej zostały umieszczone w folderze _tests_.
+
+.Folder zawierający testy dla aplikacji mobilnej
+image::../images/tests_chapter/mobileapp_tests.png[]
+
+Testy umieszczone zostały w 3 katalogach:
+
+* components- Podobnie jak w przypadku aplikacji frontendowej, w tym folderze znalazły się testy odpowiedzialne za
+sprawdzanie poszczególnych komponentów.
+
+.Folder zawierający testy związane z komponentami
+image::../images/tests_chapter/mobileapp_components_folders.png[]
+
+* concern- W tym katalogu znalazła się np. deklaracja komponentu pomocniczego, czy też atrap.
+
+.Folder zawierający m.in. atrapy oraz komponent pomocniczy
+image::../images/tests_chapter/mobileapp_concern_folder.png[]
+
+* utils- Znalazły się tutaj pliki testujące np. konwersję tablic na zmienne tekstowe, czy też generujące parę kluczy.
+
+.Folder zawierający testy związane z generowaniem par kluczy oraz konwersją.
+image::../images/tests_chapter/mobileapp_utils_folder.png[]
+
+Przykładowy test odpowiedzialny za generowanie komponentu prezentuje się następująco:
+
+.Przykładowy test z pliku SignedInRouter.test.tsx
+[source, javascript]
+----
+include::../../../mobileapp/tests/components/routers/SignedInRouter.test.tsx[lines=20..35]
+----
+
+Mimo że aplikacja mobilna jest najmniejszą częścią projektu, to pokrycie jej kodu testami wyniosło poniżej 50%. Wynik
+ten nie może zbytnio dziwić, ponieważ na implementację tego elementu poświęcono stosunkowo najmniej czasu.
+Napisanych zostało 13 przypadków testowych oraz 39 testów co jest przyzwoitym wynikiem.
+
+.Raport z pokrycia testami kodu aplikacji mobilnej
+image::../images/tests_chapter/test_coverage_mobileapp.png[]
 
 === Testy E2E
+
+Obecnie istnieją narzędzia, które umożliwiają napisanie testów sprawdzających, jak aplikacja funkcjonuje jako całość,
+bez konieczności robienia tego manualnie. Zespół zdecydował się na utworzenie również testów tego typu.
```

Comment:
brakuje mi opisu stacka, tego że jest cypress z pluginem cucmberowym. Mozesz napisać kilka zdań po co to, czym jest BDD i tak dalej.

Reply:
ok

Reply:
dodałem trochę tekstu